rotkehlchen/api/rest.py

```diff
@@ -3703,17 +3704,17 @@ def get_pnl_reports(
         return api_response(process_result(result_dict), status_code=HTTPStatus.OK)
 
     def get_report_data(self, filter_query: ReportDataFilterQuery) -> Response:
-        with_limit = False
-        entries_limit = -1
-        if self.rotkehlchen.premium is None:
-            with_limit = True
-            entries_limit = FREE_PNL_EVENTS_LIMIT
+        entries_limit, has_premium = get_user_limit(
+            premium=self.rotkehlchen.premium,
+            limit_type='pnl_events_limit',
+        )
 
         dbreports = DBAccountingReports(self.rotkehlchen.data.db)
         try:
             report_data, entries_found, entries_total = dbreports.get_report_data(
                 filter_=filter_query,
-                with_limit=with_limit,
+                with_limit=not has_premium,
+                limit=entries_limit,
```

Comment:
the with limit for what I can see is applied in `_get_reports_or_events_maybe_limit` but do we ever want to not have a limit? I feel that this argument is not needed unless you find a case for it